clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -71,6 +71,10 @@ static RegisterType getRegisterType(ResourceClass RC) {
   llvm_unreachable("unexpected ResourceClass value");
 }
 
+static RegisterType getRegisterType(const HLSLAttributedResourceType *ResTy) {
```

Comment:
You might consider renaming one of these functions, maybe add a "FromResourceType" to this new one.